### ISSUE TITLE
object-property-newline enforce either multiline or single line taking into account braces

### ISSUE THREAD

**Issue by gcmarques** [eslint/eslint#13189]
**What rule do you want to change?**
object-property-newline

**Does this change cause the rule to produce more or fewer warnings?**
More warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**
We'd add a new option. This new option would change the behavior of `allowAllPropertiesOnSameLine`. The braces would also need to be in the same line or have newlines.

a. If `{ "allowAllPropertiesOnSameLine": false }`: the behavior is the same.
b. If `{ "allowAllPropertiesOnSameLine": true }`: either all properties and opening bracket are followed by newlines or none of them is followed by newline.

The option name could be called `consistentBraces`.

**Please provide some example code that this change will affect:**
```js
/*
 eslint object-property-newline: ["error", {
   "allowAllPropertiesOnSameLine": true,
   "consistentBraces": true
 }]
*/

// Wrong
const foo = {
  a, b, c, d
};
```

**What does the rule currently do for this code?**
Currently it's allowed for `{ "allowAllPropertiesOnSameLine": true }`.

**What will the rule do after it's changed?**
```js
/*
 eslint object-property-newline: ["error", {
   "allowAllPropertiesOnSameLine": true,
   "consistentBraces": true
 }]
*/

// Wrong
const foo = {
  a, b, c, d
};

// Correct because opening bracket and all properties are on the same line.
const foo = { a, b, c, d };

// Correct because opening bracket and all properties are followed by newlines.
const foo = {
  a,
  b,
  c,
  d,
};
```

**Are you willing to submit a pull request to implement this change?**
Yes

**Previous discussions**
https://github.com/eslint/eslint/issues/13184

**Comment by mdjermanovic** [comment]
Thanks for the proposal!

I support an option to allow properties on same line only if `{` and `}` are on that line.

Just one detail. I don't think the additional option should enforce newlines after `{` and before `}`, because that's the responsibility of `object-curly-newline`.

For the `object-property-newline` rule, the following should be still valid, regardless of the new option:

```js
const foo = { a,
  b };

const bar = { 
  a,
  b };

const baz = { a,
  b 
};
```

So maybe `consistentBraces` isn't the best name for this option.

**Comment by gcmarques** [comment]
@mdjermanovic I see, it makes sense. I agree with the statement below:
> I support an option to allow properties on same line only if { and } are on that line.

 What name do you suggest?

**Comment by mdjermanovic** [comment]
I'm not sure what would be a good name for this option.

Alternatively, it could be an option for itself (not an option that modifies the other option), like `"allowSingleLine"`.

**Comment by CareLuLu-Gabriel** [comment]
`allowSingleLine: true` would allow if both brackets and properties are on the same line and not affect any other behavior. Is that what you mean?

**Comment by kaicataldo** [comment]
The team has made the difficult decision to freeze stylistic rules (please see https://eslint.org/blog/2020/05/changes-to-rules-policies for more details), and we unfortunately won't be able to accept this proposal.

Thanks for contributing to ESLint and for being willing to implement your proposal.
